File: Server/src/models/log.ts
import mongoose, {Document, Schema} from 'mongoose';

export interface ILog extends Document {
  temperature: number,
  airHumidity: number,
  soilHumidity: number,
  plantId: Schema.Types.ObjectId
}

const LogSchema: Schema = new Schema(
    {
      temperature: {
        type: Number,
        required: true,
      },
      air_humidity: Number,
      soil_humidity: Number,
      plantId: {
        type: Schema.Types.ObjectId,
        ref: 'Plant',
      },
    },
    {
      timestamps: true,
    },
);

export default mongoose.model<ILog>('Log', LogSchema);

File: App/App/Models/Log.cs
namespace App.Models
{
    public class Log
    {
        public double Temperature { get; set; }
        public double AirHumidity { get; set; }
        public double SoilHumidity { get; set; }
        public string PlantId { get; set; }
        public override string ToString()
        {
            return this.Temperature.ToString() + this.AirHumidity.ToString() + this.SoilHumidity.ToString() + this.PlantId;
        }
    }
}
